content/accessibility/button-purpose.mdx

```diff
@@ -30,37 +30,53 @@ When reviewing buttons on a page, ensure the following are true:
-- Because `aria-label` doesn’t _supplement_ visible labels but rather _supplants_ them, when a button has a visible label, include that label in its accessible name. A best practice is to have the text of the label at the start of the name.
+- When a button doesn’t have a visible label (for example: an [IconButton](https://primer.style/view-components/components/iconbutton)), provide an accessible name. Refer to [ARIA 14: Using `aria-label` to provide an invisible label where a visible label cannot be used](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA14).
+- Because `aria-label` doesn’t _supplement_ visible labels but rather _supplants_ them, when a button has a visible label, include that label in its accessible name. A good practice is to have the text of the label at the start of the name.
 - Don’t use `aria-label` when its content would be identical to a button’s visible label.
 - Don’t reuse the same (visible) label or (invisible) name for buttons which perform different actions.
 
 ## Examples
 
-### 👎 Bad: Redundant `aria-label` attribute
-
-The button below has a redundant and unnecessary `aria-label` attribute:
-
-```HTML
-<button type="button" aria-label="OK">OK</button>
-```
-
-### 👎 Bad: Missing name
-
-The button below is missing a name:
-
-```HTML
-<button type="button"><img src="https://avatars.githubusercontent.com/u/3104489?s=32"></button>
-```
-
-### 👎 Bad: Meaningless name, reused for unrelated buttons
-
-Although each button in the example below performs a different action, they all have the same name: ❌ (“cross mark”). This name does not describe the action that occurs when a given button is activated, nor does it describe the action’s context.
+<DoDontContainer>
+  <Do>
+    <Code className="language-html">
+      {`<button type="button">OK</button>`}
+    </Code>
+    <Caption>The button does not have a redundant `aria-label` attribute.</Caption>
+  </Do>
+  <Dont>
+    <Code className="language-html">
+      {`<button type="button">OK</button>`}
+    </Code>
+    <Caption>The button has a redundant and unnecessary `aria-label` attribute.</Caption>
+ </Dont>
+</DoDontContainer>
+
+
+<DoDontContainer>
+  <Do>
+    <Code className="language-html">
+      {`<button type="button" aria-label="smockle’s profile"><img alt="smockle" src="https://avatars.githubusercontent.com/u/3104489?s=32"></button>`}
+    </Code>
+    <Caption>The button should have an accessible name.</Caption>
+  </Do>
+  <Dont>
+    <Code className="language-html">
+      {`<button type="button"><img src="https://avatars.githubusercontent.com/u/3104489?s=32"></button>`}
+    </Code>
+    <Caption>The button should not be missing a name.</Caption>
+ </Dont>
+</DoDontContainer>
+
+
+### Meaningful and unique names
```

Comment:
Can this section likewise be presented in a `<DoDontContainer>` or is the code too much for it?

I'd like to retain some kind of visual indicator of which example to refer to and which example to avoid.